scripts/Invoke-AuthorizedAppDeployment.ps1

```
@@ -0,0 +1,170 @@
+  param($Row,[string]$TargetTempRoot)
+  $source = Resolve-SasInstallerSourcePath -Row $Row
+  $actualHash = (Get-FileHash -LiteralPath $source -Algorithm SHA256).Hash.ToLowerInvariant()
+  if ($actualHash -ne ([string]$Row.ExpectedSha256).ToLowerInvariant()) { throw "HASH_MISMATCH expected $($Row.ExpectedSha256) actual $actualHash" }
+  if (-not (Test-SasSafeDeploymentCleanupPath -Path $TargetTempRoot -DeploymentId $DeploymentId)) { throw "Unsafe cleanup/staging root refused: $TargetTempRoot" }
+  $session = New-PSSession -ComputerName ([string]$Row.TargetHostname)
+  try {
+    Invoke-Command -Session $session -ScriptBlock { param($p) New-Item -Path $p -ItemType Directory -Force | Out-Null } -ArgumentList $TargetTempRoot
+    $targetInstaller = Join-Path $TargetTempRoot ([System.IO.Path]::GetFileName($source))
+    Copy-Item -LiteralPath $source -Destination $targetInstaller -ToSession $session -Force
+    $remote = Invoke-Command -Session $session -ScriptBlock {
+      param($Installer,$Args,$TempRoot,$DeploymentId)
```

PSScriptAnalyzer (1.25.0)

[warning] The Variable 'Args' is an automatic variable that is built into PowerShell, assigning to it might have undesired side effects. If assignment is not by design, please use a different name.

(PSAvoidAssignmentToAutomaticVariable)